src/types/presets.ts

```diff
@@ -65,4 +67,11 @@ export interface PresetOptions {
       maxDuration: number;
     };
   };
+  firebase: {
+    /**
+     * Firebase functions v2 (2nd gen) Configuration
+     * @see https://firebase.google.com/docs/reference/functions/2nd-gen/node/firebase-functions.https.httpsoptions
+     */
+    httpRequestOptions: HttpsOptions;
```

Comment:
We need to prefix this with version (`v2`)

Reply:
@pi0 do you want to go with:

```ts
firebase: {
  v2: {
    httpRequestOptions: HttpsOptions
  }
}
```

OR

```ts
firebase: {
  v2HttpRequestOptions: HttpsOptions
}
```

Reply:
OR

```
firebaseV2: {
  httpRequestOptions: HttpsOptions
}
```

Reply:
There is a PR for adding another option that will affect both firebase and firebase-v2 presets which has been proposed as:

```
firebase: {
  serverFunctionName: string
}
```

Incase that sways the decision 😉

Reply:
Talking with @Hebilicious, i think `v2: {}` as namespace would be safest for longer run 👍🏼  (we might later introduce additional top level shortcuts for common options for better DX)

Reply:
Awesome 👍 I've updated this in [nest httpRequestOptions under the v2 key in the firebase preset options](https://github.com/unjs/nitro/pull/1142/commits/efc5bd0c0107c020388da5598a7b2112f8229cfe)